crates/net/downloaders/src/bitfinity_evm_client.rs

```diff
@@ -347,54 +367,115 @@ impl BitfinityEvmClient {
         self
     }
 
+    /// Check if the RPC endpoint is producing new blocks
+    async fn is_producing_blocks(client: &EthJsonRpcClient<ReqwestClient>) -> bool {
```

Comment:
This check should be simpler, compare the last block timestamp with the current time, if it is older than X minutes then try the fallback address

Reply:
Sorry, I don't understand how these replies are linked to my comment.